### PR TITLE
masm: added get_exemptions_points procedure

### PR DESCRIPTION
**Pull request by hackaugusto**
follow up to https://github.com/0xPolygonMiden/air-script/pull/320

This adds a `get_exemption_points` procedure which relies on the pre computed generator from the FRI stdlib.

### CODE REVIEW

File: codegen/masm/src/config.rs [hunk 1]
```diff
@@ -51,6 +51,9 @@ pub struct CodegenConfig {
     /// Memory range used to store exponentiations of Z, each address contains one point to be used
     /// on the evaluation of each periodic polynimal.
     pub z_exp_address: u32,
+
+    /// Memory position of the domain generator
+    pub domain_generator_address: u32,
```
Comment (bobbinth) [inline codegen/masm/src/config.rs:56]: This is the LDE domain generator, right? I may be good to differentiate between LDE and trace domain generator (the relationship is $g_{lde}^3 = g_{trace}$, I believe).

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one non-blocking comment inline.

File: codegen/masm/src/config.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply codegen/masm/src/config.rs:56]: It is $g_{lde}^{2^3} == g_{trace}$, basically  $g_{lde}^{blowup} == g_{trace}$

Comment (bobbinth) [reply codegen/masm/src/config.rs:56]: ah - yes! Sorry for the confusion!

**Review by Al-Kindi-0 — COMMENTED** [review]
Looks great, thank you!
As @bobbinth suggested, it makes sense to have `TRACE_DOMAIN_GENERATOR` and `LDE_DOMAIN_GENERATOR`.

File: codegen/masm/src/config.rs (hunk 1 above)
Comment (hackaugusto) [reply codegen/masm/src/config.rs:56]: > This is the LDE domain generator, right?

This is the trace domain generator. I will add a `trace_` prefix.

**Comment by hackaugusto** [comment]
Requested another review to make sure the change cleared any confusions.

File: codegen/masm/src/codegen.rs [hunk 2]
```diff
@@ -397,6 +397,30 @@ impl<'ast> Backend<'ast> {
         Ok(())
     }
 
+    /// Emits code for the procedure `get_exemptions_points`.
+    ///
+    /// Generate code to push the exemptions point to the top of the stack.
```
Comment (Al-Kindi-0) [inline codegen/masm/src/codegen.rs:402]: minor nit: "... push the exemption points..."

File: codegen/masm/src/codegen.rs [hunk 3]
```diff
@@ -458,6 +482,12 @@ impl<'ast> Backend<'ast> {
         self.writer.drop();
         self.writer.drop();
     }
+
+    /// Emits code to load `g`, the trace domain generator.
```
Comment (Al-Kindi-0) [inline codegen/masm/src/codegen.rs:486]: should we also add a suffix to g here?

**Review by Al-Kindi-0 — APPROVED** [review]
Looks great, thank you!

File: codegen/masm/src/codegen.rs (hunk 3 above)
Comment (hackaugusto) [reply codegen/masm/src/codegen.rs:486]: I can change from `g` to something else, I'm not sure what would be a good name though. Any suggestions?

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**